src/components/HdResponsive.vue

```diff
@@ -18,42 +16,46 @@ export default {
     return {
       matches: {},
       indeterminate: !matchMediaAvailable,
+      listeners: {},
     };
   },
   watch: {
     breakpoints() {
-      this.setResponsiveFlags();
+      this.setListeners();
     },
   },
-  created() {
-    this.setResponsiveFlags();
-  },
-  mounted() {
-    this.addResizeEvents();
-  },
-  beforeDestroy() {
-    this.removeResizeEvents();
+  beforeMount() {
+    this.setListeners();
```

Comment:
Why did you go for `beforeMount` instead of `created` in this case?

Reply:
Does it make any big difference?

Reply:
Depends on your use case, but you decided to go from `created` to `beforeMount` so I'm wondering why...

Reply:
@drapisarda ?

Reply:
It just sounded better

Reply:
If that is the case, I would keep it in `created` hook.